src/main/java/org/mule/extension/http/internal/request/profiling/tracing/HttpRequestCurrentSpanCustomizer.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.extension.http.internal.request.profiling.tracing;
+
+import static java.lang.String.valueOf;
+import static org.mule.runtime.http.api.domain.HttpProtocol.HTTP_0_9;
+import static org.mule.runtime.http.api.domain.HttpProtocol.HTTP_1_0;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.extension.http.internal.listener.profiling.tracing.HttpListenerCurrentSpanCustomizer;
+import org.mule.runtime.http.api.domain.HttpProtocol;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.sdk.api.runtime.source.DistributedTraceContextManager;
+import org.slf4j.Logger;
+
+import java.net.URI;
+
+/**
+ * A customizer for the current span for HTTP Requests
+ *
+ * @since 1.8.0
+ */
+public class HttpRequestCurrentSpanCustomizer extends HttpCurrentSpanCustomizer {
```

Comment:
These customizers are applied only once? Because if they're applied several times, you could pre-calculate all those values.
If they're applied only once, just ignore the suggestion

Reply:
No, they depend on some dynamic items both in the request and the listener.